destination.go

```diff
@@ -1,77 +1,199 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package vectorstore
 
 //go:generate paramgen -output=paramgen_dest.go DestinationConfig
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
 	"github.com/conduitio/conduit-commons/config"
 	"github.com/conduitio/conduit-commons/opencdc"
 	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/sashabaranov/go-openai"
 )
 
 type Destination struct {
 	sdk.UnimplementedDestination
 
 	config DestinationConfig
+	client *openai.Client
 }
 
+//go:generate paramgen -output=paramgen_dest.go DestinationConfig
+
 type DestinationConfig struct {
-	// Config includes parameters that are the same in the source and destination.
-	Config
-	// DestinationConfigParam must be either yes or no (defaults to yes).
-	DestinationConfigParam string `validate:"inclusion=yes|no" default:"yes"`
+	// APIKey is the OpenAI api key to use for the api client.
+	APIKey string `json:"api_key" validate:"required"`
+
+	// VectorStoreID is the id of the vector store to write records into.
+	VectorStoreID string `json:"vector_store_id" validate:"required"`
 }
 
 func NewDestination() sdk.Destination {
-	// Create Destination and wrap it in the default middleware.
-	return sdk.DestinationWithMiddleware(&Destination{}, sdk.DefaultDestinationMiddleware()...)
+	disable := false
+	return sdk.DestinationWithMiddleware(&Destination{}, sdk.DefaultDestinationMiddleware(sdk.DestinationWithSchemaExtractionConfig{
+		PayloadEnabled: &disable,
```

Comment:
you can do `lang.Ptr(false)` maybe? from conduit-commons/lang

Reply:
That was the one yeah, I did not remember where was it.